content/post/altswiftui-travel/index.md

```diff
@@ -0,0 +1,73 @@
+
+
+## AltSwiftUI in Rakuten Travel
+
+AltSwiftUI’s development started at the end of 2019, with the main purpose of taking on new projects while renewing the way we design software, improve the speed and quality of our development, and be more futureproof. During this journey, we tackled multiple challenges by comparing what SwiftUI originally offers with what our app needed, and the open source nature of this framework allowed us to improve it by adding features that would allow us to meet our app expectations. 
+
+_Rakuten Travel Premium Club_ is an invitation only app built with AltSwiftUI.
+
+### Simplified Architecture
+
+One of the pillars that gave us a strong advantage in terms of design and development, is the ability to simplify our app’s architecture. In contrast to using design patterns such as MVVM or VIPER, which usually tend to increase the number of things a developer needs to worry about, we could transition to a MV (model view) design with AltSwiftUI at its core. 
+
+The core principles of this MV design is that developers should focus more on what they want to build, rather than how. By clearly delimiting what is part of the UI (view layer) and what is part of our business logic/data (model layer), we were able to also easily visualize our tasks and improve development efficiency.
+
+![MV architecture](mv-architecture.png)
+
+The view layer consists of components that interact with the user. This includes the actual view structure, its local state, navigation, gestures, animations, and so on. We define all of this in a single layer by the use of declarative programming.
+
+The model layer consists of operations that retrieve, process and store data that are part of our business domain. At the same time, it exposes this data to the view layer as transparently as possible for a simple and smooth integration.
+
+The framework’s ability to integrate view state with data, including automatic updates, is what allow us to remove intermediate layers that introduce complexity and would often be the cause of bugs. Instead, the framework handles these intermediate layers without needing to add extra code.
+
+---
+
+These are the good and the bad from choosing this approach:
+
+__Pros__
+
+- Once we got used to it, our development speed increased drastically.
+- We could more easily improve our app visuals and interactions.
+- We could make changes on demand in AltSwiftUI depending on the app needs, without needing to worry about OS deployment version.
+- It became easier to request UI improvements to other developers during peer review and testing due to simpler code.
+- We could adventure with more ease of mind to prototype new and original features.
+
+__Cons__
+
+- Continuously work to improve AltSwiftUI on top of our regular tasks demanded effort from our part (specially at the start of the framework's development).
+- It required our developers to learn the combined features of SwiftUI and AltSwiftUI.
```

Comment:
the pros & cons list should use a consistent style - some bullet points use passive voice, othors use active voice, yet others use potential forms and then there's a mix of past and present tense...

An attempt at a more consistent style:

```
__Pros__

- Our development speed increased drastically.
- We can more easily improve our app visuals and interactions.
- We don't need to worry about OS deployment version anymore.
- Code Review is easier due to simpler code, this allows us to request UI improvements to other developers during peer review.
- We can experiment quickly, create prototypes new and original features.

__Cons__

- We maintain AltSwiftUI on top of our regular tasks. That was especially demanding at the start of the framework's development.
- Our developers need to learn the combined features of SwiftUI and AltSwiftUI.
```